dbt/adapters/athena/impl.py

```diff
@@ -236,15 +235,45 @@ def _get_one_catalog(
         schemas: Dict[str, Optional[Set[str]]],
         manifest: Manifest,
     ) -> agate.Table:
-        kwargs = {"information_schema": information_schema, "schemas": schemas}
-        table = self.execute_macro(
-            GET_CATALOG_MACRO_NAME,
-            kwargs=kwargs,
-            # pass in the full manifest so we get any local project
-            # overrides
-            manifest=manifest,
-        )
+        conn = self.connections.get_thread_connection()
+        client = conn.handle
 
+        with boto3_client_lock:
+            glue_client = client.session.client("glue", region_name=client.region_name, config=get_boto3_config())
+
+        catalog = []
+        paginator = glue_client.get_paginator("get_tables")
+        for schema, relations in schemas.items():
+            for page in paginator.paginate(DatabaseName=schema, MaxResults=100):
+                for table in page["TableList"]:
+                    if table["Name"] in relations:
```

Comment:
Right now I am retrieving all the tables from the database. For a few tables, it could be inefficient. I know the Glue API provides an [Expression](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue/client/get_tables.html#get_tables) parameter which is a regex, which is optimal for a few tables but is harder for a large number of tables.
What do you think @nicor88 @jessedobbelaere ?

Reply:
I think that using python string comparison on name is the most effective way.
Therefore given a db we fetch all tables, please remember to use pagination, then you filter afterwards in memory. If you use right python structures it will be fast

Reply:
I really didn't look yet the code :)